arithmetization/src/test/java/net/consensys/linea/zktracer/instructionprocessing/createTests/trivial/RootLevel.java

```diff
@@ -0,0 +1,193 @@
+    program
+        .push(0xadd7) // salt
+        .push(1) // size
+        .push(0) // offset
+        .push(1) // value
+        .op(CREATE2);
+
+    run(program);
+  }
+
+  @ParameterizedTest
+  @MethodSource("createParametersForEmptyCreates")
+  void rootLevelCreateTests(
+      CreateType createType,
+      ValueParameter valueParameter,
+      OffsetParameter offsetParameter,
+      boolean revert) {
+
+    BytecodeCompiler program = BytecodeCompiler.newProgram();
+    genericCreate(program, createType, valueParameter, offsetParameter, SizeParameter.ZERO, salt01);
+
+    if (revert) {
+      program.push(0).push(0).op(REVERT);
+    }
+
+    run(program);
+  }
+
+  @ParameterizedTest
+  @EnumSource(WhenToTestParameter.class)
+  void rootLevelCreate2AndExtCodeHash(WhenToTestParameter when) {
+
+    int storageKey = 0;
+    BytecodeCompiler program = BytecodeCompiler.newProgram();
+    precomputeDeploymentAddressOfEmptyInitCodeCreate2(program, salt01);
+    storeAt(program, storageKey);
+
+    if (when == BEFORE || when == BEFORE_AND_AFTER) {
+      loadFromStorage(program, storageKey);
+      program.op(EXTCODEHASH); // we expect to see 0
+    }
+
+    genericCreate(
+        program,
+        CreateType.CREATE2,
+        ValueParameter.ONE,
+        OffsetParameter.ZERO,
+        SizeParameter.ZERO,
+        salt01);
+
+    if (when == BEFORE || when == BEFORE_AND_AFTER) {
+      loadFromStorage(program, storageKey);
+      program.op(EQ); // we expect to have the result be true here
+      loadFromStorage(program, storageKey);
+      program.op(EXTCODEHASH); // we expect to see KECCAK(( ))
+    }
+
+    run(program);
+  }
+
+  private static Stream<Arguments> createParametersForEmptyCreates() {
+
+    final List<ValueParameter> valueParameters = List.of(ValueParameter.ZERO, ValueParameter.ONE);
+
+    List<Arguments> arguments = new ArrayList<>();
+
+    for (CreateType createType : CreateType.values()) {
+      for (ValueParameter valueParameter : valueParameters) {
+        for (OffsetParameter offsetParameter : OffsetParameter.values()) {
+          arguments.add(Arguments.of(createType, valueParameter, offsetParameter, true));
+          arguments.add(Arguments.of(createType, valueParameter, offsetParameter, false));
+        }
+      }
+    }
+
+    return arguments.stream();
+  }
+
+  public static void genericCreate(
+      BytecodeCompiler program,
+      CreateType type,
+      ValueParameter valueParameter,
+      OffsetParameter offsetParameter,
+      SizeParameter sizeParameter,
+      String salt) {
+
+    if (type == CreateType.CREATE2) program.push(salt);
+
+    switch (sizeParameter) {
+      case ZERO -> program.push(0);
+      case TWELVE -> program.push(12);
+      case THIRTEEN -> program.push(13);
+      case FOURTEEN -> program.push(14);
+      case THIRTY_TWO -> program.push(0x20);
+      case MSIZE -> program.op(MSIZE);
+      case MAX -> program.push("ff".repeat(32));
+    }
```

Comment:
Here you may consider to do the following.

Modify the enum to:

```
public enum SizeParameter {
  ZERO(0),
  TWELVE(12), // - 3 - 1
  THIRTEEN(13), // - 3 + 0
  FOURTEEN(14), // - 3 + 1
  THIRTY_TWO(32),
  MSIZE,
  MAX;

 public int getValue() {
        if (this == MSIZE || this == MAX) {
            throw new UnsupportedOperationException("...");
        }
        return this.value;
    }

  public boolean isAnyOf(SizeParameter... sizeParameters) {
    for (SizeParameter sizeParameter : sizeParameters) {
      if (this == sizeParameter) {
        return true;
      }
    }
    return false;
  }

  public boolean willRaiseException() {
    return this.isAnyOf(MAX);
  }
}
```
Then, the code above would become:

```
 switch (sizeParameter) {
      case MSIZE -> program.op(MSIZE);
      case MAX -> program.push("ff".repeat(32));
      deafult -> program.push(sizeParameter.getValue();
    }
```
However, it is not a big deal to leave it as it is, but this could be a good approach for the future.

Reply:
Another approach may be using `Either`, where you basically either store an `int` or another type, that could be an `enum`, but it is probably not worth facing that complexity for this simple scenario.

Reply:
also, what translates to an integer may have a more meaningful name rather than ZERO, TWELVE etc.